### ISSUE TITLE
Is there an API for CE?

### ISSUE THREAD

**Issue by FireCubez** [cheat-engine/cheat-engine#811]
I want to modify the memory of another program from [Rust](https://rust-lang.org) and I can't figure out how to do it with the windows API, so I thought about just making CE do it. Is it possible to use CE from an API? I saw [this issue](https://github.com/cheat-engine/cheat-engine/issues/98) but I know absolutely nothing about C# so I couldn't transfer it to Rust. Any help would be appreciated.

**Comment by Symbai** [comment]
https://github.com/fenix01/cheatengine-library

**Comment by FireCubez** [comment]
That seems dead

**Comment by cheat-engine** [comment]
You can tell ce to open a command pipe, and then issue lua commands using your program to the listening pipe of CE

Of course might be easier to just do everything including your interface inside CE

**Comment by FireCubez** [comment]
What do you mean by that? Can you make interfaces in CE?

**Comment by cheat-engine** [comment]
Yes, with Ce you can design forms, or create them using code in lua (createForm() etc..)
Then you can put code under the events of the controls (e.g button OnClick event) and in there use lua to write the code. E.g create a memscan object and make it do a scan, and then based on the results use writeInteger or any other write operation to do what you wish.

these links contain more information:
https://forum.cheatengine.org/viewforum.php?f=125
https://forum.cheatengine.org/viewforum.php?f=126
https://forum.cheatengine.org/viewforum.php?f=130
